python/examples/sample-selection.py

```diff
@@ -0,0 +1,113 @@
+# the names of the labels must be a subset of the names of the
+# samples produced by the calculator. You can see the default set
+# of names with:
+print("samples names:", descriptor.sample_names)
+
+# Now we can use these names for selection.
+# In this case, only samples  matching the labels in this selection
+# will be used
+selection = Labels(
+    names=["structure"],
+    values=np.array([[0], [2], [3]]),
+)
+
+reference_descriptor = calculator.compute(frames, selected_samples=selection)
+
+# We can compare obtained features matrix with the matrix, which we
+# will get by direct selection of atomic environments from the
+# full matrix.
+
+# obtain indices of corresponding atoms
+indices = [[i for i in range(len(frames[0].atoms))]]
+indices.append(
+    [
+        len(frames[0].atoms) + len(frames[1].atoms) + i
+        for i in range(len(frames[2].atoms))
+    ]
+)
+indices.append(
+    [
+        len(frames[0].atoms) + len(frames[1].atoms) + len(frames[2].atoms) + i
+        for i in range(len(frames[2].atoms))
+    ]
+)
+indices = np.array(indices).reshape(
+    -1,
+)
+
+reference_descriptor.keys_to_samples("species_center")
+reference_descriptor.keys_to_properties(["species_neighbor_1", "species_neighbor_2"])
+descriptor.keys_to_samples("species_center")
+descriptor.keys_to_properties(["species_neighbor_1", "species_neighbor_2"])
+
+reference_features = reference_descriptor.block(0).values
+features_direct = descriptor.block(0).values
+features_direct = features_direct[indices]
+print("Are they are equal?", np.array_equal(reference_features, features_direct))
+
+# If we want to select not only numbers of structures, but also
+# numbers of atoms, we can do the following (to select the first atoms
+# in the first two structures)
+selection = Labels(
+    names=["structure", "center"],
+    values=np.array([[0, 0], [1, 0]]),
+)
+
+reference_descriptor = calculator.compute(frames, selected_samples=selection)
+reference_descriptor.keys_to_samples("species_center")
+reference_descriptor.keys_to_properties(["species_neighbor_1", "species_neighbor_2"])
+
+# the final matrix will have 2 rows, which is what we expected.
+print(
+    "shape of first block of descriptor:", {reference_descriptor.block(0).values.shape}
+)
+
+# The previous selection method uses the same selection for all blocks.
+# If you can to use different selection for different blocks, you should
+# use a `TensorMap` to create your selection
+
+descriptor = calculator.compute(frames)
+reference_descriptor = calculator.compute(frames, selected_samples=selection)
+
+descriptor_for_comparison = calculator.compute(
+    frames, selected_samples=reference_descriptor
+)
+print("shape of first block initially:", {descriptor.block(0).values.shape})
+print(
+    "shape of first block of reference:", {reference_descriptor.block(0).values.shape}
+)
+print(
+    "shape of first block after selection:",
+    {descriptor_for_comparison.block(0).values.shape},
+)
```

Comment:
This does demonstrate the ability to use TensorMap, but feel a bit forced. I don't have a good idea on how to replace it, so maybe we should leave this out of this example? Something similar is being done for the properties selection.

Reply:
On the one hand yes, on the other hand, I personally find it very annoying when some tutorial doesn't show all the functionality. It's helpful at least to show how the function works in the case of `TensorMap`